jetty-util/src/main/java/org/eclipse/jetty/util/resource/MemoryResource.java

```diff
@@ -0,0 +1,193 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.util.resource;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+
+import org.eclipse.jetty.util.BufferUtil;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * A Memory only Resource with static content.
+ */
+public class MemoryResource extends Resource
+{
+    private final byte[] content;
+    private final long lastModified;
+
+    public MemoryResource(String rawContents)
```

Comment:
I would rather see this named `BytesResource` and take a `byte[]` and/or a `ByteBuffer` - it's what the implementation is after all and won't assume UTF-8 encoding of the String.
But then, I frankly don't see how this is any better than `XmlConfiguration(String)` - this class won't solve any of the problems that that constructor had.

Reply:
Fixed